distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,412 @@
+        } catch (Exception e) {
+            log.error("Error while waiting for login ", e);
+            throw new CarbonToolsIntegrationTestException("Error while waiting for login ", e);
+        }
+
+    }
+
+    /**
+     * This method is to check whether server is down or not
+     *
+     * @param automationContext - AutomationContext
+     * @param portOffset        - port offset
+     * @return boolean - if server is down true : else false
+     */
+    public static boolean isServerDown(AutomationContext automationContext, int portOffset)
+            throws InterruptedException {
+        boolean isPortOpen = true;
+        long startTime = System.currentTimeMillis();
+        // Looping the isPortOpen method, waiting for a while  to check the server is down or not
+        while (isPortOpen && (System.currentTimeMillis() - startTime) < TIMEOUT) {
+            isPortOpen = ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset);
+            if (isPortOpen) {
+                Thread.sleep(1000); // waiting 1 sec to check isPortOpen again
+            }
+        }
+        return !isPortOpen;
+    }
+
+    /**
+     * This method to check file has created or not
+     *
+     * @param filePathString - file path
+     * @return - if file created true else false
+     */
+    public static boolean waitForFileCreation(String filePathString) {
+        boolean isFileCreated = false;
+        long startTime = System.currentTimeMillis();
+        while ((System.currentTimeMillis() - startTime) < TIMEOUT) {
+            File createdFile = new File(filePathString);
+            if (createdFile.exists() && !createdFile.isDirectory()) {
+                isFileCreated = true;
+                break;
+            }
+        }
+        return isFileCreated;
+    }
+
+    /**
+     * This method is to return operating system
+     *
+     * @return if current os is windows return true : else false
+     */
+    public static String getCurrentOperatingSystem() {
+        return System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_OS_NAME).toLowerCase();
+    }
+
+    /**
+     * This method is to get carbon home.
+     *
+     * @param context - AutomationContext
+     * @return - carbon home
+     * @throws CarbonToolsIntegrationTestException - Error while setup carbon home from carbon zip file
+     */
+    public static String getCarbonHome(AutomationContext context)
+            throws CarbonToolsIntegrationTestException {
+        try {
+            if (carbonHome != null) {
+                return carbonHome;
+            }
+            String carbonZip = System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_CARBON_ZIP_LOCATION);
+            CarbonServerManager carbonServerManager = new CarbonServerManager(context);
+            carbonHome = carbonServerManager.setUpCarbonHome(carbonZip);
+            return carbonHome;
+        } catch (Exception ex) {
```

Comment:
I have already fixed it in pull request https://github.com/wso2/carbon4-kernel/pull/153